curator-framework/src/main/java/org/apache/curator/framework/imps/CreateBuilderImpl.java

```diff
@@ -751,7 +748,7 @@ public String forPath(String path) throws Exception {
 
     private static String getProtectedPrefix(String protectedId)
```

Comment:
This method should be public as well. Also, move the whole bit (all 4 APIs into a helper util class in the root).